bigquery/src/main/java/org/apache/iceberg/gcp/bigquery/BigQueryMetastoreClient.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.gcp.bigquery;
+
+import com.google.api.services.bigquery.model.Dataset;
+import com.google.api.services.bigquery.model.DatasetList.Datasets;
+import com.google.api.services.bigquery.model.DatasetReference;
+import com.google.api.services.bigquery.model.Table;
+import com.google.api.services.bigquery.model.TableList.Tables;
+import com.google.api.services.bigquery.model.TableReference;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * A client of Google BigQuery Metastore functions over the BigQuery service. Uses the Google
+ * BigQuery API.
+ */
+public interface BigQueryMetastoreClient {
+
+  /**
+   * Creates and returns a new dataset.
+   *
+   * @param dataset the dataset to create
+   */
+  Dataset create(Dataset dataset);
+
+  /**
+   * Returns a dataset.
+   *
+   * @param datasetReference full dataset reference
+   */
+  Dataset load(DatasetReference datasetReference);
+
+  /**
+   * Deletes a dataset.
+   *
+   * @param datasetReference full dataset reference
+   */
+  void delete(DatasetReference datasetReference);
+
+  /**
+   * Updates parameters of a dataset or adds them if did not exist, leaving already-existing ones
+   * intact.
+   *
+   * @param datasetReference full dataset reference
+   * @param parameters metadata parameters to add
+   * @return dataset after patch
+   */
+  Dataset setDatasetParameters(DatasetReference datasetReference, Map<String, String> parameters);
```

Comment:
I would probably also update the naming here and for the remove and omit the `Dataset` from the method name

Reply:
Done